lib/twingly/url.rb

```diff
@@ -109,6 +109,8 @@ def try_addressable_normalize(addressable_uri)
       end
 
       def valid_hostname?(hostname)
+        return false if hostname.nil?
+
         # No need to check the TLD, the public suffix list does that
         labels = hostname.split(DOT)[0...-1].map(&:to_s)
```

Comment:
Just some alternatives. I think current is just as good as them, but options are always fun!

To avoid work later on, not sure if `hostname` can ever be `""` though:

```suggestion
        return false if hostname.to_s.empty?

        # No need to check the TLD, the public suffix list does that
        labels = hostname.split(DOT)[0...-1].map(&:to_s)
```

or even

```suggestion
        # No need to check the TLD, the public suffix list does that
        labels = hostname.to_s.split(DOT)[0...-1].map(&:to_s)
```

I used to like the, latter, "benign value approach", but not sure I do these days :)

Reply:
I thought of doing `to_s` (and guard against the empty string) but opted not to because the code already handles empty string. Sure, avoiding unnecessary work could be good. I will leave it to someone else to [profile](https://github.com/twingly/twingly-url#profiling) if they want. :)

Reply:
We use to use guard clauses in situations like these. I'm fine with both the committed solution and the first suggestion here, so let's just go with what's already committed :D